etc/cmake/options.cmake

```
@@ -162,6 +162,11 @@ if(OT_DIAGNOSTIC)
     target_compile_definitions(ot-config INTERFACE "OPENTHREAD_CONFIG_DIAG_ENABLE=1")
 endif()
 
+option(OT_DIAG_NATIVE_RCP "enable native diag commands on rcp" OFF)
```

since it's more than just diag.
```suggestion
option(OT_APP_RCP_CLI "enable RCP cli app" OFF)
```